core/bin/external_node/src/config/mod.rs

```diff
@@ -742,70 +750,55 @@ pub(crate) struct ExternalNodeConfig {
 }
 
 impl ExternalNodeConfig {
-    /// Loads config from the environment variables and
-    /// fetches contracts addresses from the main node.
-    pub async fn collect() -> anyhow::Result<Self> {
-        let required = envy::prefixed("EN_")
-            .from_env::<RequiredENConfig>()
-            .context("could not load external node config")?;
-
-        let optional = envy::prefixed("EN_")
-            .from_env::<OptionalENConfig>()
-            .context("could not load external node config")?;
-
+    /// Loads config from the environment variables and fetches contracts addresses from the main node.
+    pub async fn new(
+        required: RequiredENConfig,
+        optional: OptionalENConfig,
+        main_node_client: &BoxedL2Client,
```

Comment:
The clients are provided so that they are not initialized multiple times (at least the L2 client is supposed to be a singleton).